src/mechanics/cell/Jacobian.h

```diff
@@ -21,53 +21,71 @@ class Jacobian
             switch (globalDimension)
             {
             case 1:
-                std::tie(mInvJacobian, mDetJacobian) = CalculateFixedSize<1>(nodeValues, derivativeShapeFunctions);
+            {
+                Eigen::Matrix<double, 1, 1> jacobian = CalculateFixedSize<1,1>(nodeValues, derivativeShapeFunctions);
+                mJacobian = jacobian;
+                mInvJacobian = jacobian.inverse();
+                mDetJacobian = jacobian.determinant();
                 break;
+            }
             case 2:
-                std::tie(mInvJacobian, mDetJacobian) = CalculateFixedSize<2>(nodeValues, derivativeShapeFunctions);
+            {
+                Eigen::Matrix<double, 2, 2> jacobian = CalculateFixedSize<2,2>(nodeValues, derivativeShapeFunctions);
+                mJacobian = jacobian;
+                mInvJacobian = jacobian.inverse();
+                mDetJacobian = jacobian.determinant();
                 break;
+            }
             case 3:
-                std::tie(mInvJacobian, mDetJacobian) = CalculateFixedSize<3>(nodeValues, derivativeShapeFunctions);
+            {
+                Eigen::Matrix<double, 3, 3> jacobian = CalculateFixedSize<3,3>(nodeValues, derivativeShapeFunctions);
+                mJacobian = jacobian;
+                mInvJacobian = jacobian.inverse();
+                mDetJacobian = jacobian.determinant();
                 break;
             }
+            }
         }
-        else
+        else if (globalDimension == interpolationDimension + 1)
```

Comment:
We have this `if`s that each contain another switch. So optionally, can you think of a suitable split into methods? Maybe even using template specialization?